src/hotspot/share/runtime/threadSMR.cpp

```diff
@@ -193,29 +193,36 @@ class ThreadScanHashtable : public CHeapObj<mtThread> {
   }
 
   int _table_size;
-  // ResourceHashtable SIZE is specified at compile time so our
-  // dynamic _table_size is unused for now; 1031 is the first prime
-  // after 1024.
-  typedef ResourceHashtable<void *, int, &ThreadScanHashtable::ptr_hash,
-                            &ThreadScanHashtable::ptr_equals, 1031,
-                            ResourceObj::C_HEAP, mtThread> PtrTable;
-  PtrTable * _ptrs;
+  // Note: I want this to be mtThread, but that causes linkage errors:
+  //
+  //   Undefined symbols for architecture x86_64:
+  //   "BasicHashtable<(MEMFLAGS)2>::free_buckets()", referenced from:
+  //   ThreadsSMRSupport::free_list(ThreadsList*) in threadSMR.o
+  //   ThreadsSMRSupport::is_a_protected_JavaThread(JavaThread*) in threadSMR.o
+  //   "BasicHashtable<(MEMFLAGS)2>::new_entry(unsigned int)", referenced from:
+  //   KVHashtable<void*, bool, (MEMFLAGS)2, &(unsigned int primitive_hash<void*>(void* const&)), &(bool primitive_equals<void*>(void* const&, void* const&))>::add_if_absent(void*, bool, bool*) in threadSMR.o
+  //
```

Comment:
We don't need these details in the comment.

Reply:
True, but they served their purpose! I'm going to check out the fix
that @robehn posted about how to get past this compilation issue.

Reply:
I don't think you should have these comments either.  They don't really help me understand why you get this compilation error.  It seems like a TODO.  Maybe the mtThread instantiation is missing from hashtable.cpp?